From 09bee32e85a5995076c01b64dfa3db0c820ba72f Mon Sep 17 00:00:00 2001
From: Wayne Adams <wadams@davey.com.au>
Date: Mon, 28 Oct 2019 08:35:10 +1100
Subject: [PATCH] enable DT overlays support

---
 arch/arm/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index d6b1629a00..27396da15c 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -440,6 +440,8 @@ config ARCH_MVEBU
 	select DM_SPI_FLASH
 	select OF_CONTROL
 	select OF_SEPARATE
+	select OF_LIBFDT
+	select OF_LIBFDT_OVERLAY
 	select SPI
 	imply CMD_DM
 
-- 
2.23.0.windows.1

